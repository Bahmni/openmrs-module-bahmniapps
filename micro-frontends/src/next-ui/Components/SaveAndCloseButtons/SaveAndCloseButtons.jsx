import React from "react";
import propTypes from "prop-types";
import {Button} from "carbon-components-react";
import "../../../styles/carbon-conflict-fixes.scss";
import "../../../styles/carbon-theme.scss";
import "../../../styles/common.scss";
import { FormattedMessage } from "react-intl";

export const SaveAndCloseButtons = (props) => {
    const { onSave, onClose, isSaveDisabled, primaryButtonText= <FormattedMessage id={'SAVE'} defaultMessage={'Save'}/> } = props;
    return (
        <div className="footer">
            <Button kind="secondary" data-testid="cancel" onClick={onClose}>
                <span><FormattedMessage id={'CANCEL'} defaultMessage={'Cancel'}/></span>
            </Button>
            <Button kind="primary" onClick={onSave} disabled={isSaveDisabled}>
                { primaryButtonText }
            </Button>
        </div>
    );
}

SaveAndCloseButtons.propTypes = {
    onSave: propTypes.func.isRequired,
    onClose: propTypes.func.isRequired,
    isSaveDisabled: propTypes.bool.isRequired,
    primaryButtonText: propTypes.string
}
export default SaveAndCloseButtons;
