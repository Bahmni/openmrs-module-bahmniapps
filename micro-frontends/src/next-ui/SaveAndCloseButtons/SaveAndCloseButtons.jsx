import React from "react";
import propTypes from "prop-types";
import {Button} from "carbon-components-react";
import "../../styles/carbon-conflict-fixes.scss";
import "../../styles/carbon-theme.scss";
import "../../styles/common.scss";
import { FormattedMessage } from "react-intl";

export const SaveAndCloseButtons = (props) => {
    const { onSave, onClose, isSaveDisabled } = props;
    return (
        <div className="footer">
            <span>
                <Button kind="secondary" data-testid="cancel" onClick={onClose}>
                    <span><FormattedMessage id={'APPOINTMENT_CREATE_CANCEL'} defaultMessage={'Cancel'}/></span>
                </Button>
            </span>
            <span>
                <Button kind="primary" onClick={onSave} data-testid="check-and-save" disabled={isSaveDisabled}>
                        <span>
                        <FormattedMessage id={'SAVE'} defaultMessage={'Save'}/>
                    </span>
                </Button>
            </span>
        </div>
    );
}

SaveAndCloseButtons.propTypes = {
    onSave: propTypes.func.isRequired,
    onClose: propTypes.func.isRequired,
    isSaveDisabled: propTypes.bool.isRequired,
}
export default SaveAndCloseButtons;