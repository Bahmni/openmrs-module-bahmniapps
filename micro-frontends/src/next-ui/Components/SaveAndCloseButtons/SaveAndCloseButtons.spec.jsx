import React from "react";
import { render, screen } from "@testing-library/react";
import SaveAndCloseButtons from "./SaveAndCloseButtons.jsx";

describe("SaveAndCloseButtons", () => {
    const onSave = jest.fn();
    const onClose = jest.fn();

    it("should render the component", () => {
        const { container } = render(<SaveAndCloseButtons onSave={onSave} onClose={onClose} isSaveDisabled={false} />);
        expect(container).toMatchSnapshot();
    });

    it("should render the component with disabled save button", () => {
        render(<SaveAndCloseButtons onSave={onSave} onClose={onClose} isSaveDisabled={true} />);
        const saveButton = screen.getByText("Save");
        expect(saveButton.getAttribute("disabled")).not.toBeNull();
    });

    it('should call onSave when save button is clicked', () => {
        render(<SaveAndCloseButtons onSave={onSave} onClose={onClose} isSaveDisabled={false} />);
        const saveButton = screen.getByText("Save");
        saveButton.click();
        expect(onSave).toHaveBeenCalledTimes(1);
    });

    it('should call onCancel when cancel button is clicked', () => {
        render(<SaveAndCloseButtons onSave={onSave} onClose={onClose} isSaveDisabled={false} />);
        const cancelButton = screen.getByText("Cancel");
        cancelButton.click();
        expect(onClose).toHaveBeenCalledTimes(1);
    });
});
