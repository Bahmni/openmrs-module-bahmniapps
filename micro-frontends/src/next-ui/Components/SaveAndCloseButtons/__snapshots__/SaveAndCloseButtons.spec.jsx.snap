// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveAndCloseButtons should render the component 1`] = `
<div>
  <div
    class="footer"
  >
    <button
      class="bx--btn bx--btn--secondary"
      data-testid="cancel"
      tabindex="0"
      type="button"
    >
      <span>
        Cancel
      </span>
    </button>
    <button
      class="bx--btn bx--btn--primary"
      tabindex="0"
      type="button"
    >
      Save
    </button>
  </div>
</div>
`;
